redis/_compat.py

```diff
@@ -15,8 +20,12 @@ def select(rlist, wlist, xlist, timeout):
         while True:
             try:
                 return _select(rlist, wlist, xlist, timeout)
-            except InterruptedError:
-                continue
+            except InterruptedError as e:
+                # Python 2 does not define InterruptedError, instead
+                # try to catch an OSError with errno == EINTR == 4.
+                if hasattr(e, 'errno') and (e.errno == 4 or e.errno is None):
+                    continue
+                raise e
```

Comment:
If we just `raise` (without the `e`), we'll get the original traceback. I'm not 100% certain, but I believe on some Python versions `raise e` would raise the exception starting there rather than where the exception originally took place.